src/main/java/com/nike/cerberus/operation/composite/RotateCertificatesOperation.java

```diff
@@ -50,42 +49,49 @@ public RotateCertificatesOperation(CloudFormationService cloudFormationService,
 
     @Override
     protected List<ChainableCommand> getCompositeCommandChain(RotateCertificatesCommand compositeCommand) {
-        List<ChainableCommand> commandList = new LinkedList<>();
-
-        if (environmentConfig.isGenerateKeysAndCerts()) {
-            commandList.add(ChainableCommand.Builder.create().withCommand(new GenerateCertificateFilesCommand()).build());
-        }
-
-        commandList.addAll(Lists.newArrayList(
+        return Lists.newArrayList(
                 // Add the cert and key files to S3
-                ChainableCommand.Builder.create().withCommand(new UploadCertificateFilesCommand()).build(),
+                ChainableCommand.Builder.create().withCommand(new UploadCertificateFilesCommand())
+                        .withAdditionalArg(UploadCertificateFilesCommandParametersDelegate.CERT_PATH_LONG_ARG)
+                        .withAdditionalArg(compositeCommand.getUploadCertificateFilesCommandParametersDelegate()
+                                .getCertPath().toString())
+                        .build(),
+
                 // Update the Load Balancer stack to use the new cert
                 ChainableCommand.Builder.create().withCommand(new UpdateStackCommand())
                         .withAdditionalArg(UpdateStackCommand.STACK_NAME_LONG_ARG)
                         .withAdditionalArg(Stack.LOAD_BALANCER.getName())
                         .build(),
+
                 // Generate new CMS config that points to the new cert
                 ChainableCommand.Builder.create().withCommand(new UpdateCmsConfigCommand()).build(),
+
                 // Do a rolling reboot of the management service
                 ChainableCommand.Builder.create().withCommand(new RebootCmsCommand()).build(),
+
                 // Delete all certs except the latest (there should just be the one)
                 ChainableCommand.Builder.create().withCommand(new DeleteOldestCertificatesCommand()).build()
-        ));
-        return commandList;
+        );
     }
 
     @Override
     public boolean isRunnable(RotateCertificatesCommand command) {
+        return areRequiredStacksPresentToRotateCertificates(cloudFormationService, environmentMetadata);
+    }
+
+    public static boolean areRequiredStacksPresentToRotateCertificates(CloudFormationService cloudFormationService,
+                                                                       EnvironmentMetadata environmentMetadata) {
+
         boolean isRunnable = true;
         String environmentName = environmentMetadata.getName();
 
         if (! cloudFormationService.isStackPresent(Stack.LOAD_BALANCER.getFullName(environmentName))) {
-            log.error("The load-balancer stack must be present in order to rotate certificates");
+            System.err.println("The load-balancer stack must be present in order to rotate certificates");
```

Comment:
Why System.err instead of log.error?

Reply:
Was trying to avoid code duplication with a static method.
Will copy code for now and re-visit a better way to avoid code duplication later.